### ISSUE TITLE
Dashboard charts fail at random after upgrade to 0.32.x

### ISSUE THREAD

**Issue by rbrthogan** [metabase/metabase#9784]
Some charts in dashboards fail with message "there was a problem displaying this chart" after upgrading to 0.32.2 from 0.31.2

Which chart that is effect seems to be random, if I refreshed the dashboard it can fix the issue on one chart but appear on another chart.

I'm using 3 filter values (id, date/single, and category). It appear to be an issue with populating these parameters 

```
{:status :failed,                                                                                                      
 :class java.util.concurrent.ExecutionException,                                                                                                                                     
 :error "java.sql.SQLNonTransientException: [Amazon][JDBC](10900) Not all parameters have been populated.",


...


:parameters                                                                                                                                                                        
  [{:type "id", :target ["variable" ["template-tag" "cohort_id"]], :value "412046733659798348"}                                                                                      
   {:type "date/single", :target ["variable" ["template-tag" "start_date"]], :value "2019-03-29"}                                                                                    
   {:type "category", :target ["variable" ["template-tag" "level_1_name"]], :value "Weddings"}],                                                                                    
  :middleware nil,                                                                                                                                                                   
  :cache-ttl nil,                                                                                                                                                                    
  :info                                                                                                                                                                              
  {:executed-by 194,                                                                                                                                                                 
   :context :question,                                                                                                                                                               
   :card-id 2146,                                                                                                                                                                    
   :dashboard-id nil,                                                                                                                                                                
   :query-hash [-119, -58, 36, -120, 49, -50, -11, 15, -22, 124, -71, 119, -3, -46, -49, -27, -9, -6, -19, -68, 125, -58, 70, -52, 43, -117, -72, 101, -109, 50, -102, -49],         
   :query-type "native"}}}

```

- Browser: Chrome 73.0.3683.86 (Official Build) (64-bit)
- OS: OS X 10 Mojave 10.14.3
- DBs: Amazon Redshift (+others not used here)
- Metabase version: ~~0.32.2~~ 0.32.4
- Metabase hosting environment: Linux
- Metabase internal database: H2

**Comment by flamber** [comment]
Hi @rbrthogan 
Could you try 0.32.4? There was several issues fixed within a very short period on 0.32.x.
If you're using a reverse proxy (like Nginx), then you might be seeing issue 9693.
By the way, you shouldn't be using H2 in production - migrate to Postgres or MariaDB/MySQL:
https://www.metabase.com/docs/latest/operations-guide/start.html#migrating-from-using-the-h2-database-to-mysql-or-postgres

**Comment by rbrthogan** [comment]
My mistake, we were actually running 0.32.4.

And yes, migration to postrgres is on the to do list :)

**Comment by lukeman** [comment]
You might try again with 0.32.5: https://github.com/metabase/metabase/releases/tag/v0.32.5
Two of the changes look related and it seems to have fixed my issues so far.

**Comment by camsaul** [comment]
Duplicate of #9693 fixed in 0.32.5

**Comment by rbrthogan** [comment]
The issue persists in 0.32.5, but happy for discussion to continue on #9693

**Comment by jianbowan** [comment]
@camsaul 
Hi, when I upgraded to 0.32.8, this problem seems to have not been resolved.
![image](https://user-images.githubusercontent.com/34364752/57714743-bdf3d380-76a7-11e9-8adf-16bebabae03b.png)


------------
Browser: Chrome 73.0.3683.86(64-bit)
OS: Windows 10
DBs: Spark
Metabase version: 0.32.8
Metabase hosting environment: Linux
Metabase internal database: postgrel

**Comment by dgleba** [comment]
I upgraded from 25.x to 32.8. I have this issue as well. I still have the 25.x running. 

My 32.8 is a docker setup where I used a mysqldump from the 25.x db. 
First, I started 32.8 and let it create a blank db. 
I then imported 25.x db and then pointed 32.8 at that db.
It started OK, but I have the issue - There was a problem displaying this chart.
When I open the question from the dashboard, the question works. Just the dashboard doesn't show the questions randomly.
